app-tasks/rf/src/rf/commands/process_upload.py

```diff
@@ -124,20 +134,48 @@ def process_upload(upload_id):
 
         generate_tasks = upload.annotationProjectId is not None and upload.generateTasks
         if generate_tasks:
-            [
-                update_annotation_project(
-                    upload.annotationProjectId, scene.ingestLocation.replace("%7C", "|"))
-                for scene in created_scenes
-            ]
-    except:
+            try:
+                [
+                    update_annotation_project(
+                        upload.annotationProjectId, scene.ingestLocation.replace("%7C", "|"))
+                    for scene in created_scenes
+                ]
+            except Exception as e:
```

Comment:
I think this whole exception handling should be doable inside the for comprehension that does the annotation project update with a lot less ceremony / duplication of the datamodel. If we do it here instead we have to maintain both copies, and nothing breaks until UAT / Joe using the app (which is frequent now but will become less so) to tell us that our python copy is out-of-date.

This call to `.void` goes from `IO[Option[Unit]]` -> `IO[Unit]` -- the `Option` tells us whether we successfully inserted the task. If it's a `None`, we failed -- and we can do the update in that job without having to do the rest of the ceremony around updating it here.

https://github.com/raster-foundry/raster-foundry/blob/develop/app-backend/batch/src/main/scala/groundwork/CreateTaskGrid.scala#L79

Reply:
Oh no I had a shower thought that says this is pretty much not gonna work, because even if we can handle all the cases here the python script still has to send an update, so all the datamodel duplication is still necessary :(

I'll call out this comment in the "choose a side!" issue but you can feel free to ignore for now